### DISCUSSION TITLE
COOKIE_DOMAIN not valid

### DISCUSSION THREAD

**Discussion by Aerglonus** [Zibbp/ganymede#45] [Q&A]
Hi,

Ganymede is running without problems but I noticed that the COOKIE_DOMAIN variable keeps printing this and I have to login every time I refresh the page, my question is, does the base URL has to explicitly be `ganymede.lonusu.com` in order for the cookie domain to be valid ? 

> ganymede-api | 2022/10/10 02:18:08 net/http: invalid Cookie.Domain "https://vods.lonusu.com"; dropping domain attribute
ganymede-api | 2022/10/10 02:18:39 net/http: invalid Cookie.Domain "https://vods.lonusu.com"; dropping domain attribute
ganymede-api | 2022/10/10 02:18:39 net/http: invalid Cookie.Domain "https://vods.lonusu.com"; dropping domain attribute

**Comment by Zibbp** [comment]
Hi,
Can you try dropping the `https://` from the `COOKIE_DOMAIN` variable, so it would be `vods.lonusu.com`. I believe this variable just needs the domain name and no protocol.

**Reply by Aerglonus** [reply]
So setting up the  `COOKIE_DOMAIN` without the protocol to `vods.lonusu.com` it did removed the error from the API container log but it created another problem, it doesn't prevent you to log in but when if I change to lets say the `live` page it logs me out with `GET https://api.lonusu.com/api/v1/live` and `POST https://api.lonusu.com/api/v1/auth/refresh` both `401 unauthorized` meaning the refresh token request failed, also noticed that in chrome base browsers the console doesn't show that the Cookie is being rejected at the moment of loading the site but checking the request in Network > login the response header shows both `set-cookie` with `This attempt to set a cookie via Set-Cookie header was blocked because its Domain attribute was invalid with regards to the current host url` error and Firefox does show it like this directly in the console: 

> Cookie “access-token” has been rejected for invalid domain. [login](https://api.lonusu.com/api/v1/auth/login)
Cookie “refresh-token” has been rejected for invalid domain. [login](https://api.lonusu.com/api/v1/auth/login)

And if I change the `COOKIE_DOMAIN` to `api.lonusu.com` there are no errors/crashes and the `POST https://api.lonusu.com/api/v1/auth/refresh` works properly everything works fine, but refreshing the page will ask me to log in again.

am I missing something or its not intended to keep the log session after refresh?

**Reply by Zibbp** [reply]
The session is intended to be kept. The way others that I've helped and how I wrote the [documentation](https://github.com/Zibbp/ganymede/wiki/Reverse-Proxy) for reverse proxying is slightly different than how you are doing it.

Ganymede (the frontend container) should be accessible using a subdomain such as `ganymede.domain.net` It does not need to be *ganymede*, anything will do. Then create subdomains for that subdomain such as `api.ganymede.zibbp.net` (for the backend API container) and `cdn.ganymede.zibbp.net` (for the Nginx container).

The `COOKIE_DOMAIN` should then be set to `ganymede.domain.net`.

I and others have had success configuring it this way. Browser cookies are very annoying to work with as they continue to be locked down more and more.
